core/src/main/scala/org/apache/spark/sql/hive/HiveClientUtil.scala

```diff
@@ -154,18 +154,6 @@ class HiveClientUtil(val sparkContext: SparkContext) extends Logging {
         dbDriver)
       metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME,
         Misc.SNAPPY_HIVE_METASTORE)
-    } else if (dbURL != null) {
-      logInfo(s"Using specified metastore database, dbURL = $dbURL")
-      metadataConf.setVar(HiveConf.ConfVars.METASTORECONNECTURLKEY, dbURL)
-      if (dbDriver != null) {
-        metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER,
-          dbDriver)
-      } else {
-        metadataConf.unset(
-          HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER.varname)
-      }
-      metadataConf.unset(
-        HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME.varname)
     } else {
       logInfo("Using Hive metastore database, dbURL = " +
```

Comment:
Did not remove this else since it was specifically for ExternalClusterMode. But looks like ExternalClusterMode is not much of use as discussed yesterday so can remove the else condition and probably file another issue to remove cluster modes that are not used any more (ExternalClusterMode and ExternalEmbeddedMode?)
@sumwale could you please confirm the above?